packages/react/src/ReactElementValidator.js

```diff
@@ -338,16 +340,20 @@ export function jsxWithValidation(
     const children = props.children;
     if (children !== undefined) {
       if (isStaticChildren) {
-        for (let i = 0; i < children.length; i++) {
-          validateChildKeys(children[i], type);
+        if (Array.isArray(children)) {
+          for (let i = 0; i < children.length; i++) {
+            validateChildKeys(children[i], type);
+          }
+        } else {
+          validateChildKeys(children, type);
```

Comment:
Yeah good catch here.

Reply:
On second read-through this `jsxs` is why it doesn't check here - `jsxs` should only be used if children is statically known to be an array. It doesn't make sense otherwise. I think we should throw some other warning when this happens (stop using jsxs in userland unless you know what you're doing!)

Reply:
I think the issue with that is that in the case of something like:
```
function foo() {
   const x = [<span />, <span />];
   return (
      <div>{x}</div>
   );
}
```
Theoretically, when you call this function, `div`'s children are statically known to be an array. However, babel doesn't know this, because Babel doesn't know what `x` is. Thus, we have to declare this as static. 

However, in this case:
```
function foo() {
   const x = <span />;
   return (
      <div>{x}</div>
   );
}
```
If `x` is actually a singleton, it would not be an array. Thus the extra check.

Reply:
This doesn't seem right. isStaticChildren should only be true when the transform itself creates the array, i.e. not the above example. So it would always be an array.

Reply:
Ah yes, I got confused about what `isStaticChildren` referred to. Thanks for the clarification! Updated it and also added a warning as per @rickyvetter's suggestion